pandas/io/tests/test_parsers.py

```diff
@@ -3809,13 +3809,26 @@ def test_decompression(self):
         try:
             import gzip
             import bz2
+            import zipfile
         except ImportError:
-            raise nose.SkipTest('need gzip and bz2 to run')
+            raise nose.SkipTest('need zipfile, gzip and bz2 to run')
 
         data = open(self.csv1, 'rb').read()
         expected = self.read_csv(self.csv1)
 
         with tm.ensure_clean() as path:
+            file_name = 'test_file'
+            tmp = zipfile.ZipFile(path, mode='w')
+            tmp.writestr(file_name, data)
+            tmp.close()
+
+            result = self.read_csv(path, compression='zip')
+            tm.assert_frame_equal(result, expected)
+
+            result = self.read_csv(open(path, 'rb'), compression='zip')
+            tm.assert_frame_equal(result, expected)
+
```

Comment:
need a test for multiple files in a zip (e.g. assert that the ``ValueError`` is raised)